designs/sam-debugging/local-sam-debugging.md

```diff
@@ -0,0 +1,331 @@
+# Local Debugging Experience for SAM Applications
+
+Current Status: Not Implemented
+
+## Introduction
+
+The AWS Toolkit enhances the SAM Application development experience by integrating local debug functionality into VS Code. This document outlines the available functionality.
+
+While this document's main focus is on debugging capabilities in the toolkit, there are places where the experience around invoking without the debugger (aka "running") is also discussed.
+
+Each programming language (and corresponding Lambda Runtimes) requires Toolkit support for debugging features to work. A limited selection of programming languages are supported in the Toolkit.
+
+### Terminology
+
+#### CodeLens
+
+CodeLenses are visual decorators anchored to a document location. They are used to convey information and/or provide links that trigger an action. They are a presentation-only mechanic and do not reside within a file. Additional information and examples about CodeLenses can be found [on the VS Code blog](https://code.visualstudio.com/blogs/2017/02/12/code-lens-roundup).
```

Comment:
" They are a presentation-only mechanic and do not reside within a file." this is a weird way to phrase it, maybe something like "are generated at runtime"

Reply:
Turns out the sentence is redundant, due to "visual decorators" in the first sentence.